### ISSUE TITLE
It is not possible to enter fractional semitone values in the Pitch Shift plugin setup.

### ISSUE THREAD

**Issue by DoctorBertalis** [wwmm/easyeffects#2453]
### EasyEffects Version

7.0.5

### What package are you using?

Flatpak (Flathub)

### Distribution

openSUSE Tumbleweed

### Describe the bug

As you know, I only listen to music at 432Hz, so today I tested EasyEffects with SoundTouch's pitch shifter plugin from Flatpack. In its settings there is one option in semitones. Therefore, to lower the tuning of music from 440 Hz to 432 Hz, I tried to enter the value "-0.32" in the tuning form, which corresponds to lowering the tuning by 32 cents, as I did in programs with this function implemented on the same SoundTouch, such as Maple JB for Android and Fubar2000 for Wine on Linux (for Windows). But I was not able to enter such a setting, but only it is possible to enter integer numbers of semitones... Or rather, you can enter "-0.32" or "-0,32", but they are not saved in the plugin settings, and the value in this case becomes zero... Although this possibility should be present, based on my experience using Maple and Foobar... Probably, this can be fixed. Otherwise functionality is lost!..

### Debug Log

<details><summary>Debug Log</summary><br/>

```
Paste your log here
```
</details>

**Comment by wwmm** [comment]
>but only it is possible to enter integer numbers of semitones...

As we are using the SoundTouch function `setPitchSemiTones` and it only accepts integer values I forced the same restriction on our side. I will have to see how fractional values are supposed to be set in SoundTouch. Its documentation is not very clear about some things.

**Comment by wwmm** [comment]
> it only accepts integer values

It seems `setPitchSemiTones` has an overload that accepts double values as input. Although it is not clear if fractional values are really used.

**Comment by wwmm** [comment]
I have updated the master branch with double replacing the integer inputs. The question now is if SoundTouch is actually using the extra digits or rounding them under the hoods.  I think that a Flatpak package will be generated in this link https://github.com/wwmm/easyeffects/actions/runs/5616605517 once github finishes to generate it. So if possible try to test it.

**Comment by DoctorBertalis** [comment]
OK, thank you! I will test EasyEffects as soon as possible. If to think about fractional numbers in the plugin settings, then I think this is true, because I checked the pitch shift of sound files on Android in Maple using the Pitch Detector program, and it showed a pitch of 432 Hz! On PC, I'll just turn on the track with a 440 Hz sine signal through EasyEffects and check the sound on Android in the Pitch Detector program.

**Comment by DoctorBertalis** [comment]
Today EasyEffects has been updated to version 7.0.6 and I hasten to report on the testing of the Pitch Shift plugin settings. Well, my previous words were confirmed today in practice! I entered the value "-0.32 semitones", turned on the 440 Hz track and got a signal at the output with a frequency equal to 431.92 Hz, almost 432 Hz! The exact value needed for a precise shift is a little less than 32 cents, which is why the output is such a value, but nevertheless it is a great result, which says that the fractional values ​​of the plugin settings are applicable! Thank you very much! And yes, the sound is very high quality!
